crates/bevy_render/src/view/visibility/mod.rs

```diff
@@ -452,6 +454,9 @@ pub fn check_visibility(
         for cell in &mut thread_queues {
             visible_entities.entities.append(cell.get_mut());
         }
+        if deterministic_rendering_config.stable_sort_z_fighting {
+            visible_entities.entities.sort_unstable();
```

Comment:
Quick clarification: why is `sort_unstable` being used when the sort is supposed to be a stable sort? Assuming there's a justification, this should probably be in a comment.

Reply:
`sort_unstable` is generally faster than `sort` when order of equal elements does not matter. It does not matter here because:
- equal entities are indistinguishable
- moreover, elements are unique

This is kind of common knowledge, and usually not explained. There are much more complex things happen in this function, which may need explanations.

If something may need explanation, it is the other issue: per-thread results are sorted, so we may avoid sorting the array, but run the merge of sorted per-thread results instead. But I'm not 100% sure it will be faster, and will work correctly, and explanation is tricky, so I omitted it as well.

Reply:
i agree a comment here would be useful. `if stable { sort_unstable() }` looks like a bug and will evidently cause confusion.

Reply:
```suggestion
            // We can use the faster unstable sort here because the values (`Entity`) are guaranteed to be unique
            visible_entities.entities.sort_unstable();
```

Reply:
OK then.